terraform/functions.tf

```diff
@@ -286,54 +291,130 @@ module "lambda_function-excelToJson" {
-    module.postgres.security_group_id,
-  ]
-  handler        = "index.handler"
-  architectures  = [var.lambda_arch]
-  runtime        = var.lambda_runtime
-  publish        = true
-  layers         = local.lambda_layer_arns
+  // Networking
+  attach_network_policy  = false
+  vpc_subnet_ids         = null
+  vpc_security_group_ids = null
+
+  // Permissions
+  role_permissions_boundary         = local.permissions_boundary_arn
+  attach_cloudwatch_logs_policy     = true
+  cloudwatch_logs_retention_in_days = var.log_retention_in_days
+  attach_policy_jsons               = true
+  number_of_policy_jsons            = length(local.lambda_default_execution_policies)
+  attach_policy_statements          = true
+  policy_statements = {
+    AllowDownloadExcelObjects = {
+      effect = "Allow"
+      actions = [
+        "s3:GetObject",
+        "s3:HeadObject",
+      ]
+      resources = [
+        # Path: uploads/{organization_id}/{agency_id}/{reporting_period_id}/{expenditure_category_code}/{upload_id}/{filename}.xlsm
+        "${module.reporting_data_bucket.bucket_arn}/uploads/*/*/*/*/*/*.xlsm",
+      ]
+    }
+    AllowUploadJsonObjects = {
+      effect  = "Allow"
+      actions = ["s3:PutObject"]
+      resources = [
+        # Path: uploads/{organization_id}/{agency_id}/{reporting_period_id}/{expenditure_category_code}/{upload_id}/{filename}.xlsm.json
+        "${module.reporting_data_bucket.bucket_arn}/uploads/*/*/*/*/*/*.xlsm.json",
+      ]
+    }
+  }
+
+  // Artifacts
   create_package = false
   s3_existing_package = {
     bucket = aws_s3_object.lambda_artifact-excelToJson.bucket
     key    = aws_s3_object.lambda_artifact-excelToJson.key
   }
 
-  role_name     = "lambda-role-excelToJson"
-  attach_policy = true
-  policy        = "arn:aws:iam::aws:policy/AmazonS3FullAccess"
+  // Runtime
+  handler       = var.datadog_enabled ? local.datadog_lambda_handler : "excelToJson.handler"
+  runtime       = var.lambda_runtime
+  architectures = [var.lambda_arch]
+  publish       = true
+  layers        = local.lambda_layer_arns
+  timeout       = 300 # 5 minutes, in seconds
+  memory_size   = 512 # MB
+  environment_variables = merge(local.lambda_default_environment_variables, {
+    DD_LAMBDA_HANDLER = "excelToJson.handler"
+  })
+
+  // Triggers
+  allowed_triggers = {
+    S3BucketNotification = {
+      principal  = "s3.amazonaws.com"
+      source_arn = module.reporting_data_bucket.bucket_arn
+    }
+  }
 }
 
 module "lambda_function-cpfValidation" {
-  source        = "terraform-aws-modules/lambda/aws"
-  version       = "6.5.0"
-  function_name = "cpf-validation"
+  source  = "terraform-aws-modules/lambda/aws"
+  version = "6.5.0"
+
+  // Metadata
+  function_name = "${var.namespace}-cpfValidation"
   description   = "Reacts to S3 events and validates CPF JSON files."
 
-  vpc_subnet_ids = local.private_subnet_ids
-  vpc_security_group_ids = [
-    module.lambda_security_group.id,
-    module.postgres.security_group_id,
-  ]
```

Comment:
Since the `cpfValidation ` handler doesn't seem to require DB access at the moment, we can avoid adding the Lambda function to the VPC.